src/core/simulation.jl

```diff
@@ -977,33 +1010,39 @@ function _execute!(sim::Simulation, store; cache_size_mib = 1024, kwargs...)
                         # TODO: implement some efficient way of indexing with stage name.
                         stage = get_stage(sim, stage_number)
                         stage_name = get_stage_name(sim, stage)
-                        !stage_built(stage) &&
+                        if !is_stage_built(stage)
                             error("Stage $(stage_name) status is not BUILT")
+                        end
                         stage_interval = get_stage_interval(sim, stage_name)
                         run_name = "step-$(step)-stage-$(stage_name)"
                         sim.internal.current_time = sim.internal.date_ref[stage_number]
                         sim.internal.time_step = sim.internal.time_step_ref[stage_number]
-                        sim.sequence.current_execution_index = ix
+                        # TODO: Show progress meter here
+                        get_sequence(sim).current_execution_index = ix
                         @info "Starting run $run_name $(get_current_time(sim))"
-                        raw_results_path = joinpath(
-                            sim.internal.raw_dir,
-                            run_name,
-                            replace_chars("$(get_current_time(sim))", ":", "-"),
-                        )
-                        mkpath(raw_results_path)
                         # Is first run of first stage? Yes -> don't update stage
                         TimerOutputs.@timeit RUN_SIMULATION_TIMER "Update Stage $(stage_number)" begin
                             !(step == 1 && ix == 1) && update_stage!(stage, sim)
                         end
                         TimerOutputs.@timeit RUN_SIMULATION_TIMER "Run Stage $(stage_number)" begin
                             stage_name = get_stage_name(sim, stage)
-                            run_stage(
-                                step,
-                                stage,
-                                get_current_time(sim),
-                                raw_results_path,
-                                store,
-                            )
+                            settings = get_settings(stage)
+                            # TODO: Add Fallback when run_stage fails
+                            status = run_stage!(step, stage, get_current_time(sim), store)
+                            if status == SUCESSFUL_RUN
+                            elseif !get_allow_fails(settings) && (status != SUCESSFUL_RUN)
+                                set_simulation_status!(sim, FAILED_RUN)
+                                break
+                            elseif get_allow_fails(settings) && (status != SUCESSFUL_RUN)
+                                continue
```

Comment:
Commentary: In the passing case you proceed through this block. In this case where there is an allowed failure you immediately continue the nearest for loop. A lot of logic gets skipped. If it is correct to do this then a comment is probably warranted. The logic may be unnecessarily complex.

Reply:
Yeah, if the fail is allowed, the remaining steps should be skipped. We could probably refactor the function to simplify it. I think that should be done in the next iteration